deployments/container/cleanup/cleanup.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express orimplied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+echo "install_weak_deps=False" >> /etc/dnf/dnf.conf
+rm -f /etc/dnf/protected.d/*.conf
+
+rm -f /etc/yum.repos.d/cuda.repo
+rm -f /etc/ld.so.conf.d/nvidia.conf
+
+dnf remove -y \
+    cuda* \
+    systemd
+
+dnf clean all
+rm -rf /var/cache/dnf
+
+dnf install -y microdnf
+
+microdnf remove $(rpm -q --whatrequires dnf)
+rpm -e dnf
+
+microdnf remove \
+    $(rpm -q --whatrequires /usr/libexec/platform-python) \
+    $(rpm -q --whatrequires 'python(abi)') \
+    python* \
+    dnf*
+
+microdnf remove \
```

Comment:
does this ensure there are no broken dependencies? i.e remaining packages in the cuda:12.5.0-base-ubi8 might have dependencies on packages removed with the minimal list as a reference?

Reply:
In my testing, `microdnf` still respects package dependencies, meaning that if `x` depends on `y`, one cannot remove `y`.

For example, let's say we want to remove `openssl` we see the following in the image:
```
$ microdnf remove openssl*
error: Could not depsolve transaction; 1 problem detected:
 Problem: package curl-7.61.1-34.el8.x86_64 from @System requires libcrypto.so.1.1()(64bit), but none of the providers can be installed
  - package curl-7.61.1-34.el8.x86_64 from @System requires libssl.so.1.1()(64bit), but none of the providers can be installed
  - conflicting requests
  - problem with installed package curl-7.61.1-34.el8.x86_64
```